### PR TITLE
[PR] Implement Load Module

### PR DESCRIPTION
**Pull request by kcw36**
# Description

- implement load module business logic
- Checks database for duplicate records
- Checks database if it needs to add region or state
- Basic tests for pure functionality

Closes: #42 
Closes: #47

## Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Refactoring (code improvement without functional changes)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] My code has a pylint score over 8

# Additional Details

- this module will need more testing but for a minimum requirement this is acceptable as we push for our deadline.

# Mentions
@xaviachristi
@Emily-Prust
@MariosT4
@kcw36

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+)
+
+
+def get_connection() -> Connection:
+    """Return connection to database."""
+    return connect(
+        host=ENV["DB_HOST"],
+        user=ENV["DB_USER"],
+        dbname=ENV["DB_NAME"],
+        port=int(ENV["DB_PORT"]),
```
Comment (Emily-Prust) [inline pipeline/load.py:28]: Is `int` necessary here?

File: pipeline/load.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        if col in df.columns:
+            df[col] = to_numeric(df[col], errors='coerce')
+
+    df['latitude'] = df['latitude'].round(6)   # matches DECIMAL(9,6)
+    df['longitude'] = df['longitude'].round(6)
+    df['magnitude'] = df['magnitude'].round(1)  # matches DECIMAL(3,1)
+    df['depth'] = df['depth'].round(2)         # matches DECIMAL(5,2)
+    df['cdi'] = df['cdi'].round(1)
+    df['mmi'] = df['mmi'].round(1)
+    df['dmin'] = df['dmin'].round(3)
```
Comment (Emily-Prust) [inline pipeline/load.py:66]: Good thinking as there seems to be a bit of variance here.

File: pipeline/load.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    if not new_quakes.empty:
+        logger.debug("Found new earthquake data: \n%s", new_quakes.head())
+    else:
+        logger.debug("No new earthquake data found.")
+
+    logger.info("Uploading new data to database...")
+    upload_df_to_db(db_conn, new_quakes)
+
+    logger.info("Load run successful.")
+    return new_quakes
```
Comment (Emily-Prust) [inline pipeline/load.py:206]: I think that pulling the logging out of the functions like this makes the script easier to read at the cost of a little reusability. Would you recommend pulling logging out like this or leaving it attached to the functions?

File: pipeline/load.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                        JOIN "region" USING (region_id);""")
+        quakes = DataFrame(curs.fetchall())
+    if quakes.empty:
+        quake_cols = [
+            "earthquake_id", "magnitude", "latitude", "longitude", "time", "updated",
+            "depth", "url", "felt", "tsunami", "cdi", "mmi", "nst", "sig", "net", "dmin",
+            "alert", "location_source", "magnitude_type", "state_name", "region_name",
+            "state_id", "region_id"
+        ]
+        return DataFrame(columns=quake_cols)
```
[the remark this answers is not in the document]
Comment (kcw36) [reply pipeline/load.py:48]: QA moment: Take a note and we will pursue it as a quality action next week.
